### PR TITLE
Convert to rizin APIs

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

convert to rizin APIs

- `iCj`
- `icj`
- `iij`
- `iej`
- `ihj`
- `iSSj`
- `CClj`

**Test plan (required)**

**Closing issues**

Partially addresses https://github.com/rizinorg/cutter/issues/2666

### CODE REVIEW

File: src/core/Cutter.cpp [hunk 1]
```diff
@@ -3177,20 +3234,31 @@ QList<FlirtDescription> CutterCore::getSignaturesDB()
...
-    QList<CommentDescription> ret;
-
-    for (CutterJson commentObject : cmdj("CClj")) {
-        QString type = commentObject[RJsonKey::type].toString();
-        if (type != filterType)
+    QList<CommentDescription> qList;
+    RzIntervalTreeIter it;
+    void *pVoid;
+    RzAnalysisMetaItem *item;
+    RzSpace *spaces = rz_spaces_current(&core->analysis->meta_spaces);
```
Comment (ret2libc) [inline src/core/Cutter.cpp:3241]: is this necessary? I don't see something like this done in the code of `CCl`.

File: src/core/Cutter.cpp [hunk 2]
```diff
@@ -3354,82 +3422,113 @@ QStringList CutterCore::getSectionList()
...
-    for (CutterJson segmentObject : cmdj("iSSj")) {
-        QString name = segmentObject[RJsonKey::name].toString();
-        if (name.isEmpty())
+    RzBinSection *segment;
+    RzListIter *iter;
+    QList<SegmentDescription> ret;
+    CutterRzListForeach (segments, iter, RzBinSection, segment) {
+        if (RZ_STR_ISEMPTY(segment->name)) {
             continue;
+        }
```
Comment (ret2libc) [inline src/core/Cutter.cpp]: no need

File: src/core/Cutter.cpp (hunk 1 above)
Comment (b1llow) [reply src/core/Cutter.cpp:3241]: https://github.com/rizinorg/rizin/blob/7665d0f30ef7771b305329502d94842ec4a97a62/librz/core/cmeta.c#L75-L78

maybe it's equivalent to this

Comment (ret2libc) [reply src/core/Cutter.cpp:3241]: ouch, that's a bad place actually for the filter IMHO.. but yeah, you are right then and this is fine.

File: src/core/Cutter.cpp [hunk 3]
```diff
@@ -3354,82 +3422,109 @@ QStringList CutterCore::getSectionList()
...
+    CutterRzListForeach (segments, iter, RzBinSection, segment) {
+        SegmentDescription segDesc;
+        segDesc.name = segment->name;
+        segDesc.vaddr = segment->vaddr;
+        segDesc.paddr = segment->paddr;
+        segDesc.size = segment->size;
+        segDesc.vsize = segment->vsize;
+        segDesc.perm = perms_str(segment->perm);
     }
+    rz_list_free(segments);
```
Comment (wargio) [inline src/core/Cutter.cpp:3454]: do you really need to free segments? aren't these just an RZ_BORROW ?

**Comment by wargio** [comment]
the windows build works (i gave it a spin)

**Review by XVilka — APPROVED** [review]
Tested on Windows 10, it works.

**Comment by XVilka** [comment]
@imbillow doesn't show comment in Chinese ;)
See the comment at  0x1400022c6 address:

image

**Comment by b1llow** [comment]
image

@XVilka It seems that Chinese will be encoded like this. And `CCl` does not have `pd`-like behavior.

**Comment by XVilka** [comment]
It might be a separate PR, but I think it's a bit inconvenient for the user, and we shouldn't filter the printable names in both Rizin via `CCl` and Cutter via GUI.

**Review by wargio — APPROVED** [review]

**Comment by wargio** [comment]
for me we can merge this

**Review by ret2libc — APPROVED** [review]

**Pull request merged**